File: PHP/2390. Removing Stars From a String.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    public function removeStars(string $s): string
    {
        $array = [];

        for ($i = 0; $i < strlen($s); $i++) {
            $word = $s[$i];
            if ($word === '*') {
                array_pop($array);
            } else {
                $array[] = $word;
            }
        }

        return implode('', $array);
    }
}

File: Go/2390. Removing Stars From a String.go
package main

//goland:noinspection GoUnusedFunction
func removeStars(s string) string {
	array := make([]rune, 0)

	for _, word := range s {
		if word != '*' {
			array = append(array, word)
		} else {
			array = array[:len(array)-1]
		}
	}

	return string(array)
}
